src/Form/PeriodForm.php

```diff
@@ -119,6 +120,27 @@ public function form(FormBuilderInterface $builder, array $options) : void
                 $days->setData(array_fill(0, 7, []));
             }
         });
+
+        // Reset days info language to match the selected language.
```

Comment:
Maybe something like:
```suggestion
        // Days info language needs to be reset to match the period. 
```
To comment more on the "why" than the "how".